arch/xtensa/src/esp32/esp32_spiflash.c

```diff
@@ -170,6 +183,27 @@ struct spiflash_cachestate_s
   uint32_t val[CONFIG_SMP_NCPUS];
 };
 
+#ifdef CONFIG_ESP32_SPI_FLASH_SUPPORT_PSRAM_STACK
+/* SPI flash work operation arguments */
+
+struct spiflash_work_arg
+{
+  enum spiflash_op_code_e op_code;
+
+  struct
+  {
+    struct esp32_spiflash_s *priv;
+    uint32_t addr;
+    uint8_t *buffer;
+    uint32_t size;
+  } op_arg;
+
+  int ret;
```

Comment:
```suggestion
  volatile int ret;
```

Reply:
Why here needs volatile ?

Reply:
The modification is done in another context. This is just to ensure that value is always re-read after waiting on semaphore

Reply:
Yeah, this is much important in SMP and hardware using external RAM.

Reply:
Yes, each core has its own cache of 32KB. I think the CPU core should be designed to make sure data is ready after we call memory sync instruction like `volatile`(this will be transformed into xtensa instruction  'memw').